src/logic/BrazeCards.test.ts

```diff
@@ -0,0 +1,247 @@
+                expect(result).toEqual([]);
+            });
+
+            test("returns BrazeCard instances for each of the profile badge cards in the user's cached card feed", () => {
+                jest.spyOn(appboy, 'getCachedContentCards').mockImplementation(() => {
+                    return new appboy.ContentCards(
+                        [
+                            testCard('ProfileBadge', 'card-id-1', { field1: 'value-1' }),
+                            testCard('ProfileBadge', 'card-id-2', { field1: 'another-value' }),
+                            testCard('ProfileBadge', 'card-id-3', { field2: 'yet-another-value' }),
+                        ],
+                        new Date(),
+                    );
+                });
+                const brazeCards = new BrazeCards(appboy, errorHandler);
+                const result = brazeCards.getCardsForProfileBadge();
+                const cardIds = result.map((card) => card.id);
+                expect(cardIds).toEqual(['card-id-1', 'card-id-2', 'card-id-3']);
+            });
+
+            test("uses Braze's card ID", () => {
+                fc.assert(
+                    fc.property(fc.string(), (id) => {
+                        jest.spyOn(appboy, 'getCachedContentCards').mockImplementation(() => {
+                            return new appboy.ContentCards(
+                                [testCard('ProfileBadge', id, {})],
+                                new Date(),
+                            );
+                        });
+                        const brazeCards = new BrazeCards(appboy, errorHandler);
+                        const result = brazeCards.getCardsForProfileBadge();
+                        expect(result[0].id).toEqual(id);
+                    }),
+                );
+            });
+
+            test('uses the extras record that Braze provides', () => {
+                const extrasGen = fc.dictionary(fc.string(), fc.string());
+                fc.assert(
+                    fc.property(extrasGen, (dict) => {
+                        const extras = { ...dict, slotName: 'ProfileBadge' };
+                        jest.spyOn(appboy, 'getCachedContentCards').mockImplementation(() => {
+                            return new appboy.ContentCards(
+                                [testCard('ProfileBadge', 'card-id', dict)],
+                                new Date(),
+                            );
+                        });
+                        const brazeCards = new BrazeCards(appboy, errorHandler);
+                        const result = brazeCards.getCardsForProfileBadge();
+                        expect(result[0].extras).toEqual(extras);
+                    }),
+                );
+            });
+
+            test('returns no cards if the cached feed contains only unsupported cards', () => {
+                // cards with no extras or missing the slotName, so they cannot be handled by this library
```

Comment:
Do we want to include the "no extras" case in the test below, to ensure it gets filtered out correctly? Related, should this comment be moved into the test description? It's more specific than the "unsupported cards" description we've currently got. So something like:

```
test('returns no cards if the cached feed contains only cards with no extras or missing the slotName', () => {
```